vali_objects/utils/subtensor_weight_setter.py

```diff
@@ -66,105 +73,79 @@ def set_weights(self, current_time: int = None):
                 if miner in self.metagraph.hotkeys:
                     challengeperiod_weights.append((
                         self.metagraph.hotkeys.index(miner),
-                        ValiConfig.SET_WEIGHT_MINER_CHALLENGE_PERIOD_WEIGHT
+                        ValiConfig.CHALLENGE_PERIOD_WEIGHT
                     ))
                 else:
                     bt.logging.error(f"Challengeperiod miner {miner} not found in the metagraph.")
 
-            transformed_list = checkpoint_netuid_weights + challengeperiod_weights
+            weights_list = checkpoint_netuid_weights + challengeperiod_weights
+            transformed_hotkeys = [x[0] for x in weights_list]
+
+            metagraph_remaining_hotkeys = list(set(self.metagraph.hotkeys) - set(transformed_hotkeys))
+            metagraph_remaining_scores = [(hotkey, 0) for hotkey in metagraph_remaining_hotkeys]
+
+            transformed_list = weights_list + metagraph_remaining_scores
             bt.logging.info(f"transformed list: {transformed_list}")
 
             self._set_subtensor_weights(transformed_list)
         self.set_last_update_time()
 
-    def filtered_ledger(
-        self,
-        hotkeys: List[str] = None
-    ) -> PerfLedger:
+    @staticmethod
+    def sync_ledger_positions(
+            ledger,
+            positions
+    ) -> tuple[dict[str, PerfLedger], dict[str, list[Position]]]:
         """
-        Filter the ledger for a set of hotkeys.
+        Sync the ledger and positions to ensure that the ledger and positions are in the same state.
         """
-        if hotkeys is None:
-            hotkeys = self.metagraph.hotkeys
-
-        # Note, eliminated miners will not appear in the dict below
-        ledger = self.perf_manager.load_perf_ledgers_from_disk()
+        ledger_keys = set(ledger.keys())
+        position_keys = set(positions.keys())
 
-        augmented_ledger = {}
-        for hotkey, miner_ledger in ledger.items():
-            if hotkey not in hotkeys:
-                continue
+        common_keys = ledger_keys.intersection(position_keys)
+        uncommon_keys = ledger_keys.union(position_keys) - common_keys
 
-            ledger_copy = copy.deepcopy(miner_ledger)
-            if not ledger_copy.cps:
-                continue
+        synced_ledger = {}
+        synced_positions = {}
 
-            checkpoint_filtered_elements = self._filter_checkpoint_elements(ledger_copy.cps)
-            checkpoint_meets_criteria = self._filter_checkpoint_list(checkpoint_filtered_elements)
-            if not checkpoint_meets_criteria:
-                continue
+        for hotkey in common_keys:
+            synced_ledger[hotkey] = ledger[hotkey]
+            synced_positions[hotkey] = positions[hotkey]
 
-            ledger_copy.cps = checkpoint_filtered_elements
-            augmented_ledger[hotkey] = ledger_copy
+        # if len(uncommon_keys) > 0:
```

Comment:
Is this left here intentionally?

Reply:
@jbonilla-tao @sli-tao do we think it's a good idea / necessary to coerce syncing on the positions and ledgers like this? How long does it take for the ledger to generate on the first position

Reply:
On average the time delta should be ~20 minutes between the first position and the initial perf ledger. I think upperbound could be 1 hour. I don't think forced sync is possible or needed. Can simply wait until the ledger exists.

Reply:
If the ledger and positions are out of sync would that generate incorrect miner statistics?

Reply:
think we're ok. scoring should skip over miners without both